src/Session.hs

```diff
@@ -92,10 +92,10 @@ isSuccess :: Maybe Summary -> Bool
 isSuccess = not . isFailure
 
 parseSummary :: String -> Maybe Summary
-parseSummary r = case reverse $ lines r of
```

Comment:
I think at this point we want tests for `parseSummary`.  It document's your change and makes sure that it will not get lost with a future refactoring.